aip/0131.md

```diff
@@ -85,6 +85,36 @@ paths:
   - The variable for the resource ID **must** be named `id`. Variables
     representing the ID of parent resources **must** end with `Id`.
 
+#### RAML
+
+Single-resource `GET` methods **must** be specified in accordance with the
+following RAML spec.
+
+```yaml
+/publishers/{publisherId}/books/{bookId}:
```

Comment:
This pattern conflicts with the above callout that the OAS standard specifically calls out that the final resource should simply be {id}.  

Neither I feel are technically wrong, but we should be consistent I think within the AIP as they are saying conflicting things and this isn't really a difference between OAS and RAML.

My two cents is I prefer this way since then you don't have the changing IDs for parent calls.  In the one above you'd have to do the following in RAML.

```yaml
/publishers/{id}:
  ...
/publishers/{publisherId}/books/{id}:
  ...
```

where the unique naming you'd have the following in RAML since you'd be able to nest your resources.
```yaml
/publishers/{publisherId}:
  ...
  /books/{bookId}:
    ...
    /pages/{pageId}:
      ...
```

Just something to consider.

Reply:
I intentionally did this different than the the OAS and the protobuff example, hoping to have this conversation. Thanks for noticing.

The primary benefit of using `{bookId}` instead of `{id}` is that you can define the `bookId` parameter in the `components/parameters` in OAS or as a trait (I think) in RAML. Then reusing those parameters is more DRY.

I was interested in seeing if the `{id}` requirement in protobufs is preference or backed by some precedent.